Source/WebKit/WebProcess/WebPage/WebPage.cpp

```diff
@@ -3861,6 +3861,15 @@ void WebPage::runJavaScript(WebFrame* frame, RunJavaScriptParameters&& parameter
     }
 #endif
 
+    bool shouldAllowUserInteraction = [&] {
+#if PLATFORM(COCOA)
+        if (linkedOnOrAfterSDKWithBehavior(SDKAlignedBehavior::ProgrammaticFocusDuringUserScriptShowsInputViews))
+            return true;
+#endif
+        return m_userIsInteracting;
+    }();
+
+    SetForScope userIsInteractingChange { m_userIsInteracting, shouldAllowUserInteraction };
```

Comment:
This looks fine, since it doesn't appear that this method can get called while we're inside an existing `m_userIsInteracting` scope, but I wonder if we should be more defensive by making this:

```
SetForScope userIsInteractingChange { m_userIsInteracting, m_userIsInteracting | shouldAllowUserInteraction };
```